pkg/manager/impl/execution_manager.go

```diff
@@ -234,26 +234,36 @@ func createTaskDefaultLimits(ctx context.Context, task *core.CompiledTask,
 	}
 	taskResourceLimits := runtimeInterfaces.TaskResourceSet{}
 
-	// For resource values, we prefer to use the limits set in the application config over the set resource values.
-	if !configResourceLimits.CPU.IsZero() {
+	// For resource values, we prefer to set unset limits to the request values over the config values.
+	if cpuIndex >= 0 {
+		cpuLimit = resourceRequestEntries[cpuIndex].Value
+		cpu, err := resource.ParseQuantity(cpuLimit)
+		if err != nil {
+			logger.Errorf(ctx, "Failed to parse user cpu limit from task spec [%s] with err [%+v]", cpuLimit, err)
+		} else {
+			taskResourceLimits.CPU = cpu
+		}
+	} else if !configResourceLimits.CPU.IsZero() {
 		taskResourceLimits.CPU = configResourceLimits.CPU
 	} else {
-		if cpuIndex >= 0 {
-			cpuLimit = resourceRequestEntries[cpuIndex].Value
-		}
 		cpu, err := resource.ParseQuantity(cpuLimit)
 		if err != nil {
 			logger.Errorf(ctx, "Failed to parse user cpu limit from task spec [%s] with err [%+v]", cpuLimit, err)
 		} else {
 			taskResourceLimits.CPU = cpu
 		}
 	}
-	if !configResourceLimits.Memory.IsZero() {
+	if memoryIndex >= 0 {
+		memoryLimit = resourceRequestEntries[memoryIndex].Value
+		memory, err := resource.ParseQuantity(memoryLimit)
+		if err != nil {
+			logger.Errorf(ctx, "Failed to parse user memory limit from task spec [%s] with err [%+v]", memoryLimit, err)
```

Comment:
same as above, better errors please

Reply:
code has been refactored but the full task id is in the error msg